tests/checkpatch_wrapper_test.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+. ./kw.sh --source only
+. ./src/checkpatch_wrapper.sh --source only
+
+function suite
+{
+  suite_addTest "testCheckpatch"
+}
+
+function testCheckpatch
+{
+  test -e $external_script_path/checkpatch.pl; assertEquals "File checkpatch.pl does not exist" $? 0
+  tmp_path="/tmp/kw_test"
+  mkdir -p $tmp_path && wget \
+    "https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/plain/kernel/exit.c" \
+    -q -O "${tmp_path}/exit.c"
+  [[ "$?" -ne 0 ]] && fail "Could not download test file from https://git.kernel.org/ to $tmp_path"
+  res=$(execute_checkpatch "${tmp_path}/exit.c" 2>&1 | tail -n 1);\
+    [[ "$?" -ne 0 ]] && fail "Could not execute first codestyle run on ${tmp_path}/exit.c"
+  printf ' %.0s' {1..100} >> "${tmp_path}/exit.c"
+  nres=$(execute_checkpath "${tmp_path}/exit.c" 2>&1 | tail -n 1);\
+    [[ "$?" -ne 0 ]] && fail "Could not execute second codestyle run on ${tmp_path}/exit.c"
+  [ "$res" == "$nres" ] && fail "Codestyle did not capture warnings"
+  true # Reset return value
+}
+
+. ./tests/shunit2
```

Comment:
Ideally, a unit test should not depend on external sources[1]. In other words, depends on the Internet connection and an external file is not good for tests. Additionally, I recommend you to use shunit2 functions such as "assertFalse" and "setUp/tearDown" [2]. Follow my suggestion for this part of the code:
1) Create a setUp/tearDown function for any basic setup required by your test.
2) Instead of download any file, just put it inside the test directory (e.g., tests/samples). This makes the test self-contained. In this specific case, I also recommend you to maintain the checkpatch file and replace the target used by kw.
3) Try to use the shunit2 function [2]

1. https://en.wikipedia.org/wiki/Unit_testing#Description
2. https://github.com/kward/shunit2#-function-reference